fireworks/fireworks.go

```diff
@@ -36,26 +37,45 @@ type model struct {
 	system *simulation.System
 }
 
-func Spawn(width, height int) []simulation.Particle {
+func SpawnShoot(width, height int) *simulation.Particle {
 	color := lipgloss.Color(array.Sample(colors))
-	v := float64(rand.Intn(10) + 20.0)
-
-	particles := []simulation.Particle{}
+	v := float64(rand.Intn(15) + 15.0)
 
 	x := rand.Float64() * float64(width)
-	y := rand.Float64() * float64(height)
+
+	p := simulation.Particle{
+		Physics: simulation.NewProjectile(
+			simulation.FPS(framesPerSecond),
+			simulation.Point{X: x, Y: float64(height)},
+			simulation.Vector{X: 0, Y: -v},
+			simulation.Vector(simulation.TerminalGravity),
+		),
+		Char:          lipgloss.NewStyle().Foreground(color).Render(head_char),
+		TailChar:      lipgloss.NewStyle().Foreground(color).Render(tail_char),
+		Shooting:      true,
+		ExplosionCall: SpawnExplosion,
+	}
+	return &p
+}
+
+func SpawnExplosion(x, y float64, width, height int) []*simulation.Particle {
+	color := lipgloss.Color(array.Sample(colors))
+	v := float64(rand.Intn(10) + 20.0)
+
+	particles := []*simulation.Particle{}
 
 	for i := 0; i < numParticles; i++ {
 		p := simulation.Particle{
-			Physics: harmonica.NewProjectile(
-				harmonica.FPS(framesPerSecond),
-				harmonica.Point{X: x, Y: y},
-				harmonica.Vector{X: math.Cos(float64(i)) * v, Y: math.Sin(float64(i)) * v / 2},
-				harmonica.Vector(harmonica.TerminalGravity),
+			Physics: simulation.NewProjectile(
+				simulation.FPS(framesPerSecond),
+				simulation.Point{X: x, Y: y},
+				simulation.Vector{X: math.Cos(float64(i)) * v, Y: math.Sin(float64(i)) * v / 2},
+				simulation.Vector(simulation.TerminalGravity),
```

Comment:
How come we are removing `harmonica` here?

I actually moved all the internal code to harmonica https://github.com/charmbracelet/harmonica/pull/6 for confetty to make use of it 😄

Reply:
Probably would be better to just expose the velocity and acceleration in harmonica then?

Reply:
but as I said this PR isn't complete yet so just wait while I make it work fully

Reply:
would be great if you could send PR to harmonica to add functions to get the velocity and acceleration of a projectile

Reply:
https://github.com/charmbracelet/harmonica/pull/9

Reply:
awesome

Reply:
Btw merge the harmonica PR